### PR TITLE
Add the `maximize` flag to AdamW

### PR DESCRIPTION
**Pull request by Adnios**
Related issue: https://github.com/pytorch/pytorch/issues/68052

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### CODE REVIEW

File: torch/optim/adamw.py [hunk 1]
```diff
@@ -12,11 +12,16 @@ class AdamW(Optimizer):
...
                 \: \epsilon \text{ (epsilon)}                                                    \\
-            &\hspace{13mm}      \lambda \text{(weight decay)}, \:  amsgrad                       \\
+            &\hspace{13mm}      \lambda \text{(weight decay)},  \: \textit{amsgrad},
+                \: \textit{maximize}                                                             \\
             &\textbf{initialize} : m_0 \leftarrow 0 \text{ (first moment)}, v_0 \leftarrow 0
                 \text{ ( second moment)}, \: \widehat{v_0}^{max}\leftarrow 0              \\[-1.ex]
             &\rule{110mm}{0.4pt}                                                                 \\
             &\textbf{for} \: t=1 \: \textbf{to} \: \ldots \: \textbf{do}                         \\
+
+            &\hspace{5mm}\textbf{if} \: \textit{maximize}:                                       \\
```
Comment (albanD) [inline torch/optim/adamw.py:22]: This is missing indentation in the if/else, see rendered doc: https://docs-preview.pytorch.org/70146/generated/torch.optim.AdamW.html?highlight=adamw#torch.optim.AdamW

File: torch/optim/adamw.py [hunk 2]
```diff
@@ -58,7 +65,7 @@ class AdamW(Optimizer):
     """
 
     def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-8,
-                 weight_decay=1e-2, amsgrad=False):
+                 weight_decay=1e-2, amsgrad=False, *, maximize: bool = False):
```
Comment (albanD) [inline torch/optim/adamw.py:68]: no space around the = sign here?

**Review by albanD — COMMENTED** [review]
Looks good! Thanks

File: torch/optim/adamw.py (hunk 2 above)
Comment (Adnios) [reply torch/optim/adamw.py]: Thanks for catching. Fixed.

File: torch/optim/adamw.py (hunk 1 above)
Comment (Adnios) [reply torch/optim/adamw.py:22]: Fixed. Thanks.

File: torch/optim/adamw.py (hunk 2 above)
Comment (Adnios) [reply torch/optim/adamw.py]: It seems that there must space around the `=`.
```
./torch/optim/adamw.py:68:69: E252 missing whitespace around parameter equals
./torch/optim/adamw.py:68:70: E252 missing whitespace around parameter equals
./torch/optim/_multi_tensor/adamw.py:34:69: E252 missing whitespace around parameter equals
./torch/optim/_multi_tensor/adamw.py:34:70: E252 missing whitespace around parameter equals
Please fix the above Flake8 warnings.
```

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**